### ISSUE TITLE
Libssh2_session_handshake failed. Invalid MAC received.

### ISSUE THREAD

**Issue by Himura2la** [allanrbo/filesremote#10]
![image](https://user-images.githubusercontent.com/7802808/181192776-1ac7018b-d62c-4165-bb29-24e6d79e90bc.png)

```
PS C:\Users\himura\.ssh> cat .\config
Host o2
    HostName Omega-FFFF
    User root
    IdentityFile ~/.ssh/omega-ffff_rsa

Host *
    User ansible
    IdentityFile ~/.ssh/id_ed25519
```

Sometimes it also says "Failed getting banner".

**Comment by allanrbo** [comment]
What’s the operating system and version of the server you are connecting to?

That error must be coming from libssh2, so we may need to investigate if it’s a known issue there.

**Comment by Himura2la** [comment]
I tried to connect to Debian 11 with the latest OpenSSH and to OpenWRT with dropbear, no luck. Both connections work fine with OpenSSH versions from Windows and from Debain 11 @ WSL

**Comment by allanrbo** [comment]
This is very strange. Any chance you could check if it works when using password instead of keys?

I just successfully connected to a Debian 11 box using an RSA key.

**Comment by allanrbo** [comment]
Ah. I just got a problem when trying an OpenWrt 21.02.1 router. The error I got in the status bar was "Libssh2_sftp_init failed. Timed out waiting on socket. Reconnecting...". Have you also sometimes got this?

Wireshark revealed that _something_ was being tried. There was a conversation back and forth. Libssh2's init must just have been expecting to get some message from the SSH server that it's not getting...

**Comment by allanrbo** [comment]
Another question @Himura2la , did this used to work on versions of FilesRemote prior to 1.8? I just realized that on the latest build, I was using the latest master branch of https://github.com/libssh2/ rather than a specific version. So if it worked on previous versions, maybe there's a bug in the newest Libssh2.

**Comment by Himura2la** [comment]
Looks like I can't connect using the password, it fails before prompting, event if I rename the `~/.ssh` directory. I was able to connect using password in previous version.

**Comment by allanrbo** [comment]
I was able to reproduce the "Invalid MAC received" issue for the Windows version when connecting to a Debian 11 server. Looks like it was because on my Windows machine, I had accidentally used the master branch of openssl and libssh2. I have now rebuilt using the newest newest stable release tags of these dependencies.

I have uploaded a new build, 1.10, where this is fixed.

Unfortunately, the issue with connecting to OpenWRT (Dropbear) seems to persist. Looks like it is unrelated though. Will create a separate issue for it.
